spec/classes/puppet_agent_prepare_spec.rb

```diff
@@ -6,8 +6,8 @@
 MCO_LOGFILE = '/var/log/puppetlabs/mcollective.log'
 
 describe 'puppet_agent::prepare' do
-  context 'supported operating systems' do
-    on_supported_os.each do |os, facts|
+  context 'supported operating system families' do
+    on_supported_os.to_a.uniq { |os, facts| facts[:osfamily] }.each do |os, facts|
```

Comment:
This doesn't work as expected in ruby 1.8.7:

```
irb(main):005:0>  {'debian-7' => 'debian', 'debian-6' => 'debian' }.to_a.uniq { |k,v| v }.each {|k,v| puts "#{k}=#{v}"}
debian-7=debian
debian-6=debian
=> [["debian-7", "debian"], ["debian-6", "debian"]]
```

But does in ruby 1.9 and up:

```
irb(main):001:0> {'debian-7' => 'debian', 'debian-6' => 'debian' }.to_a.uniq { |k,v| v }.each {|k,v| puts "#{k}=#{v}"}
debian-7=debian
```